### DISCUSSION TITLE
Diagram size and animations

### DISCUSSION THREAD

**Discussion by vedeja** [likec4/likec4#770] [General]
Most of my diagrams have 10-20 elements and even at 10 you begin to see degradation in performance, making zooming, click to highlight, and hovering quite slow. I have experimented a bit to try and disable the animations altogether, with some kind of improved but unsatisfying result. Also with the UX more plain and boring. Is 10 elements in a view stretching the limit? Can we optimize the rendering? Is an option to disable animation a bad idea?

As discussed in #600, v1 did improve things to the layout in general but overall the slow animations still haunt me.

**Comment by davydkov** [comment]
@vedeja, hmmm, I see degradation only in Safari (macos and ios).
Chrome, Arc, Firefox - everything runs smoothly.
Can you record a screen video and send me denis@davydkov.com?

**Reply by vedeja** [reply]
Sure! I do agree that Safari is worse than Chrome, but coming from Structurizr means that I still have this expectation that loading the diagram is slow, after which the navigation is fast.
